src/Compilers/Core/Portable/PEWriter/DebugDirectoryExtensions.cs

```diff
@@ -19,9 +22,11 @@ private static void InitializeReflection()
         {
             if (s_dataBuilderField == null)
             {
-                var type = typeof(DebugDirectoryBuilder).GetTypeInfo();
-                s_dataBuilderField = type.GetDeclaredField("_dataBuilder");
-                s_addEntry = (Action<DebugDirectoryBuilder, DebugDirectoryEntryType, uint, uint, int>)type.GetDeclaredMethod("AddEntry").CreateDelegate(typeof(Action<DebugDirectoryBuilder, DebugDirectoryEntryType, uint, uint, int>));
+                var type = typeof(DebugDirectoryBuilder);
+                s_dataBuilderField = type.GetTypeInfo().GetDeclaredField("_dataBuilder");
+                s_addEntry = (Action<DebugDirectoryBuilder, DebugDirectoryEntryType, uint, uint, int>)
+                    RuntimeReflectionExtensions.GetRuntimeMethod(type, "AddEntry", new[] { typeof(DebugDirectoryBuilder), typeof(DebugDirectoryEntryType), typeof(uint), typeof(uint), typeof(int) }).
```

Comment:
Just curious, why call GetRuntimeMethod as static instead of as extension?

Reply:
The IDE wasn't happy about it for some reason.

Reply:
GetRuntimeMethod doesn't work anyways. It only returns public method